### PR TITLE
nixos-rebuild: add lib to repl to make debugging even easier

### PR DESCRIPTION
**Pull request by SuperSandro2000**
## Description of changes

When debugging things you often need lib. I think that is probably the last commonly used arg.
Tested on my system.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by roberth** [comment]
Great!

I've added two commits to make sure it's going to return the correct variation of `lib`.

**Comment by SuperSandro2000** [comment]
Thanks! I would have probably taken a long time to do the same changes.

**Comment by SuperSandro2000** [comment]
vm tests seem to fail :/

```
error processing entry /build/root/nix/store/sz011gslhakpyd8jykijq15hqyrwfav7-rustc-1.75.0-doc, aborting
error processing entry /build/root/nix/store, aborting
error processing entry /build/root/nix, aborting
[   78.089026] reboot: Restarting system
ERROR: cptofs failed. diskSize might be too small for closure.
error: builder for '/nix/store/wl87n285zf8mn7d00v2pg49xhczvvvjz-nixos-disk-image.drv' failed with exit code 1;
```

**Comment by roberth** [comment]
Looks like it's on the edge of fitting within the allocated size. It passed on arm.

The build does pull in all of its dependencies, to make sure it can build a fresh configuration in Nix in the VM, so that might explain a large need.
Might be worth checking that the NixOS runtime closure is kinda ok, but other than that we should just increase the disk allocation.

**Comment by roberth** [comment]
It's not a blocker for this PR though, because I don't think this test actually tests `nixos-rebuild repl`. I might not have written a test at all, which is admittedly very bad of me.

**Comment by roberth** [comment]
@ofborg build nixos-rebuild
@ofborg build nixos-rebuild.tests
@ofborg build nixos-rebuild.tests.repl

File: pkgs/os-specific/linux/nixos-rebuild/test/repl.nix [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    expect_simple "nix-repl>"
+
+    send "config.networking.hostName\n"
+    expect_simple "itsme"
+
+    expect_simple "nix-repl>"
+    send "lib.version\n"
+    expect_simple ${escapeExpect (
+      # The version string is a bit different in the flake lib, so we expect a prefix and ignore the rest
+      lib.strings.substring 0 4 (lib.strings.escapeNixString lib.version))}
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/nixos-rebuild/test/repl.nix:129]: Is it safe to use lib.version here? I am not sure if that would trigger a rebuild of this test on every commit.

**Comment by SuperSandro2000** [comment]
The test failures seem unrelated. Feel free to merge merge when you are happy with it.

**Review by roberth — APPROVED** [review]
I approve of your approval :handshake:

**Comment by roberth** [comment]
Backport was not necessary, but might have been nice for facilitating further backports.

**Pull request merged**